src/components/modal/index.stories.mdx

```diff
@@ -0,0 +1,58 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs';
+import { useDisclosure, ChakraProvider } from '@chakra-ui/react';
+
+import Modal from './index';
+import ModalContent from './ModalContent';
+import ModalCloseButton from './ModalCloseButton';
+import ModalBody from './ModalBody';
+import ModalHeader from './ModalHeader';
+import ModalFooter from './ModalFooter';
+import ModalOverlay from './ModalOverlay';
+
+import ThemeProvider from '../themeProvider';
+import Divider from '../divider';
+import Button from '../button';
+import Box from '../box';
+
+<Meta title="Components/Data display/Modal" component={Modal} />
+
+export const Template = (args) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <Box>
+      <Button onClick={onOpen}>Open Modal</Button>
+      <Modal isOpen={isOpen} onClose={onClose} {...args}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Modal Title</ModalHeader>
+          <ModalCloseButton />
+          <Divider />
+          <ModalBody>
+            <Box p="4">This is modal content</Box>
+          </ModalBody>
+        </ModalContent>
+      </Modal>
+    </Box>
+  );
+};
+
+# Modal
+
+<Canvas>
+  <Story
+    name="Modal"
+    args={{ isCentered: true, size: 'lg', motionPreset: 'scale' }}
+    argTypes={{
+      size: {
+        options: ['lg', 'full'],
```

Comment:
I assume that is result of discussion with designers - to have only 2 sizes for the modal. For me `lg` is a bit strange name for the  width of modal - 460px (e.g. share modal). For me it looks like small to medium modal :).